celltype_annotation/analysis/SingleR-combining-refs.Rmd

```diff
@@ -0,0 +1,218 @@
+params:
+  s3_data_dir: "s3://nextflow-ccdl-results/scpca/processed/results/SCPCP000007" 
+  local_data_dir: "../data"
+  sample_id: "SCPCS000222"
+  library_id: "SCPCL000296"
+  s3_ref_dir: "s3://scpca-references/celltype/references"
+  local_ref_dir: "../references"
+---
+
+Results for `r params$library_id`.
+
+This notebook is focused on a preliminary analysis of using SingleR with combining references.
+I mostly followed the setup outlined in the [SingleR book section on combining multiple references with harmonizing labels](http://bioconductor.org/books/release/SingleRBook/using-multiple-references.html#using-harmonized-labels).
+To do this, I created a list of all references and then grabbed the `label.ont` from all references and provided both of those as input to `SingleR::SingleR()` to annotate a single library.
+
+## Set Up
+
+```{r}
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+  library(ggplot2)
+})
+theme_set(theme_bw())
+```
+
+```{r}
+# make local data directory if it doesn't exist
+fs::dir_create(params$local_data_dir)
+fs::dir_create(params$local_ref_dir)
+```
+
+```{r}
+# Define list of reference files that we will use for combined analysis
+refs <- c("BlueprintEncodeData", 
+          "HumanPrimaryCellAtlasData",
+          "DatabaseImmuneCellExpressionData",
+          "MonacoImmuneData")
+ref_files <- glue::glue("celldex-{refs}.rds")
+local_ref_paths <- file.path(params$local_ref_dir, ref_files) |>
+  purrr::set_names(refs)
+
+# if missing the reference files, grab them from S3
+if(!all(file.exists(local_ref_paths))){
+  # sync reference files 
+  aws_includes <- paste("--include '", ref_files, "'", sep = '', collapse = ' ')
+  sync_call <- paste('aws s3 cp', params$s3_ref_dir, params$local_ref_dir,
+                     '--exclude "*"', aws_includes,'--recursive', sep = " ")
+  system(sync_call, ignore.stdout = TRUE) 
+}
+
+# read all ref files in as a single list
+celldex_refs <- purrr::map(local_ref_paths, readr::read_rds)
+```
+
+
+```{r}
+# build path to annotated sce file 
+annotated_sce_file <- glue::glue("{params$library_id}_annotated.rds")
+local_annotated_sce_path <- file.path(params$local_data_dir, 
+                                      params$sample_id, 
+                                      annotated_sce_file)
+
+# if missing the annotated sce files, grab them from S3
+if(!all(file.exists(local_annotated_sce_path))){
+  # sync annotated SCE files 
+  aws_includes <- glue::glue('--include "*/{annotated_sce_file}"')
+  
+  sync_call <- glue::glue('aws s3 sync {params$s3_data_dir} {params$local_data_dir} --exclude "*" {aws_includes}')
+  system(sync_call, ignore.stdout = TRUE) 
+}
+
+# read in all annotated sce 
+annotated_sce <- readr::read_rds(local_annotated_sce_path)
+```
+
+## SingleR with combined reference
+
+Before providing the list of references and labels, first subset them so that they contain the same set of genes.
+
+```{r}
+# identify shared genes among all references
```

Comment:
From my read of the docs, it seemed like this may not really be a required step, but I doubt it will cause trouble.